### PR TITLE
:recycle: [llm][npu][quant] Delegate MXFP8 dense scheme to kernel and use torch.ops.npu

### PR DESCRIPTION
**Pull request by TallMessiWu**
## Motivation

Follow-up to #22352 (W8A8 MXFP8 for Qwen3 Dense on Ascend NPU). It addresses two
review comments from @TamirBaydasov that were deferred to keep that already-approved
PR stable:

- https://github.com/sgl-project/sglang/pull/22352#discussion_r3395802464 — a scheme
  should own only hardware-agnostic weight creation and delegate
  `process_weights_after_loading` / `apply` to a linear-method kernel (`self.kernel`),
  like the other ModelSlim schemes.
- https://github.com/sgl-project/sglang/pull/22352#discussion_r3395718707 — NPU ops can
  be reached through `torch.ops.npu.*`, so the top-level `import torch_npu` is unnecessary.

## Modifications

- `ModelSlimMXFP8Scheme` now keeps only `create_weights` and delegates
  `process_weights_after_loading` / `apply_weights` to `NPUMXFP8LinearMethod`
  (`self.kernel`), matching `ModelSlimW8A8Int8`.
- `NPUMXFP8LinearMethod.process_weights_after_loading` branches on the weight dtype:
  pre-quantized `float8_e4m3fn` weights (offline ModelSlim) are only re-laid-out, while
  `float16`/`bfloat16` weights (online) are quantized to MXFP8 at load time. Both paths
  converge on `weight_scale_inv`, so `apply` is shared.
- MXFP8 ops now go through `torch.ops.npu.npu_dynamic_mx_quant` /
  `torch.ops.npu.npu_quant_matmul` with standard `torch.float8_*` dtypes, matching the
  int8/int4 methods in the same file. The top-level `import torch_npu` and the
  `current_platform.is_npu()` guard are dropped from both files, keeping them importable
  on non-NPU (CUDA/CPU/AMD/XPU) CI.

No functional change: the online (`--quantization mxfp8`) and offline
(`--quantization modelslim`, `W8A8_MXFP8`) paths keep the same weight/scale layout.

## Accuracy Test / Benchmark

- `pre-commit run --all-files` passes.
- Verified end-to-end on Ascend NPU: Qwen3 dense online (`--quantization mxfp8`) and
  offline (`--quantization modelslim`) both produce coherent output; the weight/scale
  layout is unchanged, so throughput is on par with the pre-refactor baseline.

---
### CI States

Latest PR Test (Base): :x: [Run #27675484610](https://github.com/sgl-project/sglang/actions/runs/27675484610)
Latest PR Test (Extra): :x: [Run #27675484240](https://github.com/sgl-project/sglang/actions/runs/27675484240)

### CODE REVIEW

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py [hunk 1]
```diff
@@ -169,33 +166,48 @@ def create_weights(
...
+        if weight.dtype == torch.float8_e4m3fn:
+            # Offline (ModelSlim) path: weight is already MXFP8-quantised and
+            # layer.weight_scale holds the uint8 block scales [out, in/32]. Only
+            # re-layout to [in, out] / [in//64, out, 2] strided views below.
+            n_dim, k_dim = layer.weight_scale.data.shape
+            scale = layer.weight_scale.data.reshape(n_dim, k_dim // 2, 2)
+            layer.weight = Parameter(weight.transpose(0, 1), requires_grad=False)
+            layer.weight_scale_inv = Parameter(
+                scale.transpose(0, 1), requires_grad=False
+            )
```
[the remark this answers is not in the document]
Comment (TallMessiWu) [reply python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py:185]: Done in f796a23606. In the offline (ModelSlim) branch `weight_scale` is folded into `weight_scale_inv`, so I now `del layer.weight_scale` to drop the stale parameter from `named_parameters()` / `state_dict()`. `weight_scale_inv` is a view over the same storage, so this only removes the duplicate registration (no data copy).

File: python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py [hunk 2]
```diff
@@ -6,24 +6,17 @@
...
 MXFP8_BLOCK_SIZE = 32
-_FLOAT8_E8M0FNU_DTYPE = (
-    getattr(torch_npu, "float8_e8m0fnu", getattr(torch, "float8_e8m0fnu", None))
-    if _is_npu
-    else getattr(torch, "float8_e8m0fnu", None)
-)
+# NPU ops are reached via torch.ops.npu.* (registered when torch_npu is imported
+# by the runtime), so this module needs no top-level `import torch_npu` and stays
+# importable on CUDA/CPU/AMD/XPU CI.
+_FLOAT8_E8M0FNU_DTYPE = getattr(torch, "float8_e8m0fnu", None)
```
[the remark this answers is not in the document]
Comment (TallMessiWu) [reply python/sglang/srt/hardware_backend/npu/quantization/linear_method_npu.py]: `_FLOAT8_E8M0FNU_DTYPE` is now a `_get_float8_e8m0fnu_dtype()` helper called from `apply()` instead of a module-level constant. `torch.float8_e8m0fnu` is a torch-core dtype so the resolved value is identical, but reading it lazily avoids capturing it at import time and is the more robust pattern given this module is imported on non-NPU CI as well.

**Review by ping1jing2 — APPROVED** [review]

**Comment by TamirBaydasov** [comment]
LGMT
